### ISSUE TITLE
Create a Holdout Set

### ISSUE THREAD

**Issue by HarshCasper** [HakinCodes/Malaria-Detection#38]
## Type

Feature

## Description

While training, validating and optimizing our model we could over time start to overfit the validation data without realizing it. This means that the model will perform well during training but it will perform poorly on unseen data.

Create a holdout set containing 200 images. We will keep this holdout set aside and only use it at the end to check how the final model performs on unseen data. Keep the code in a `Scripts/` directory for future use.

## Tools

- Python

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/HakinCodes/Malaria-Detection/blob/master/CONTRIBUTING.md)?

Yes

**Comment by aryanVijaywargia** [comment]
I would like to work on this @HarshCasper

**Comment by BALaka-18** [comment]
@aryanVijaywargia Assigned

**Comment by macabdul9** [comment]
Holdout set will be from the same distribution, hence the performance will be the same as validation. This is the main problem of machine learning models that they do not perform well on out of the distribution data. For the demo at the client end, we can take few samples (ie: 10) from the main directory and then split the data into train, Val/test set (so that model doesn't get trained on demo data).

**Comment by HarshCasper** [comment]
I guess we can create a seperate issue for that @macabdul9

**Comment by BALaka-18** [comment]
> Holdout set will be from the same distribution, hence the performance will be the same as validation. This is the main problem of machine learning models that they do not perform well on out of the distribution data. For the demo at the client end, we can take few samples (ie: 10) from the main directory and then split the data into train, Val/test set (so that model doesn't get trained on demo data). @aryanVijaywargia @HarshCasper

@macabdul9 open a new issue for this. You'll be assigned to work on it.

**Comment by aryanVijaywargia** [comment]
I have a query. I have written a python script that generates 100 samples at random from each class and moves the images to the holdout_dataset directory. So should my pr contain both the holdout_dataset directory (containing the images) as well as the code or only the code will suffice? @BALaka-18

**Comment by BALaka-18** [comment]
> I have a query. I have written a python script that generates 100 samples at random from each class and moves the images to the holdout_dataset directory. So should my pr contain both the holdout_dataset directory (containing the images) as well as the code or only the code will suffice? @BALaka-18

@aryanVijaywargia both. The sample you created can be used for initial testing, or as an example when we document our model.

**Comment by aryanVijaywargia** [comment]
Thanks for clarifying @BALaka-18

**Comment by macabdul9** [comment]
> > Holdout set will be from the same distribution, hence the performance will be the same as validation. This is the main problem of machine learning models that they do not perform well on out of the distribution data. For the demo at the client end, we can take few samples (ie: 10) from the main directory and then split the data into train, Val/test set (so that model doesn't get trained on demo data). @aryanVijaywargia @HarshCasper
> 
> @macabdul9 open a new issue for this. You'll be assigned to work on it.

I think mentors can not contribute

**Comment by BALaka-18** [comment]
> > > Holdout set will be from the same distribution, hence the performance will be the same as validation. This is the main problem of machine learning models that they do not perform well on out of the distribution data. For the demo at the client end, we can take few samples (ie: 10) from the main directory and then split the data into train, Val/test set (so that model doesn't get trained on demo data). @aryanVijaywargia @HarshCasper
> > 
> > 
> > @macabdul9 open a new issue for this. You'll be assigned to work on it.
> 
> I think mentors can not contribute

@macabdul9 I'm sorry I forgot. Open an issue then, participants will be assigned.

**Comment by rutujadhanawade** [comment]
Is this issue open?
